### DISCUSSION TITLE
Rotate cells contents

### DISCUSSION THREAD

**Discussion by Zarko1234** [lvjr/tabularray#85] [Ideas]
Is possible to  set some row feature such that cells' contents in this row is rotated for selected angle (typically 90 degree)? If possible, how to specify such row?

Typical application of such features are column headers when narrow columns is requested. This this can be done for example by use of the `rothead` macro defined in the `makecell` package (for rotating is needed `rotating` package) for each cell separately. This is at tables with many columns quite annoying.

**Comment by lvjr** [comment]
You can write `row{1}={cmd=\rotcell}` with `makecell` package.

**Reply by kiryph** [reply]
Consider this example:
```tex
\documentclass[border=1cm]{standalone}
\usepackage{rotating}
\usepackage{makecell}
\usepackage{tabularray}

\begin{document}
\begin{tblr}{
  row{1} = {ht=2.5cm,valign=b,halign=c,cmd=\rotcell},
  hlines,vlines
}
  Alpha   & Beta  & Gamma   & Delta   \\
  Epsilon & Zeta  & Eta     & Theta   \\
  Iota    & Kappa & Lambda  & Mu      \\
  Nu      & Xi    & Omicron & Pi      \\
  Rho     & Sigma & Tau     & Upsilon \\
  Phi     & Chi   & Psi     & Omega   \\
\end{tblr}
\end{document}
```
and its output
![Screenshot 2022-08-15 at 12 58 00](https://user-images.githubusercontent.com/371756/184623734-b99c9f09-417b-4581-8557-9472178d3233.png)



which has following issues:

1. the word Alpha is broken
2. the vertical alignment is not as desired; already added `valign=b` which does not change anything
3. the horizontal alignment is not as desired; already added `halign=c` which does not change anything

How can these issues be fixed?

**Reply by kiryph** [reply]
I could fix it by using the predefined `\rothead` command and `rotheadsize` setting of the package `makecell`. I also disabled the special font with`\renewcommand\theadfont{}`:

```tex
\documentclass[border=1cm]{standalone}
\usepackage{rotating}
\usepackage{makecell}
\usepackage{tabularray}

\setlength\rotheadsize{1.25cm}
\renewcommand\theadfont{}

\begin{document}
\begin{tblr}{
  row{1} = {halign=c,cmd=\rothead},
  hlines,vlines
}
  Alpha   & Beta  & Gamma   & Delta   \\
  Epsilon & Zeta  & Eta     & Theta   \\
  Iota    & Kappa & Lambda  & Mu      \\
  Nu      & Xi    & Omicron & Pi      \\
  Rho     & Sigma & Tau     & Upsilon \\
  Phi     & Chi   & Psi     & Omega   \\
\end{tblr}
\end{document}
```


![Screenshot 2022-08-15 at 13 01 14](https://user-images.githubusercontent.com/371756/184624072-5bfb13a6-f7ae-42d6-bde9-08b124e07c72.png)

**Reply by sgolovan** [reply]
Also, `cmd=\rotatebox{90}` works as well. It never breaks the cell contents into lines though.

**Comment by Zarko1234** [comment]
Thank you very much! This works. Now I only have a wish that an example of table with rotated cell.

**Comment by Rmano** [comment]
The rothead/makecell thing works, but rotated headers are quite common, it would be nice to have a `rotate` (or similar) embedded!

**Comment by lvjr** [comment]
@Rmano There is issue #98 for this feature. Please add your comments there on how to handle different cases and how the interfaces should be.
